reconcile/utils/ocm/base.py

```diff
@@ -118,13 +126,62 @@ class OCMClusterFlag(BaseModel):
     enabled: bool
 
 
+class OCMClusterAWSOperatorRole(BaseModel):
+    id: str
+    name: str
+    namespace: str
+    role_arn: str
+    service_account: str
+
+
+class OCMAWSSTS(OCMClusterFlag):
+    role_arn: Optional[str]
+    support_role_arn: Optional[str]
+    oidc_endpoint_url: Optional[str]
+    operator_iam_roles: Optional[list[OCMClusterAWSOperatorRole]]
+    instance_iam_roles: Optional[dict[str, str]]
+    operator_role_prefix: Optional[str]
+
+
 class OCMClusterAWSSettings(BaseModel):
-    sts: Optional[OCMClusterFlag]
+    sts: Optional[OCMAWSSTS]
 
     @property
     def sts_enabled(self) -> bool:
         return self.sts is not None and self.sts.enabled
 
+    @property
+    def aws_account_id(self) -> str:
+        return get_account_uid_from_arn(self.account_roles[0])
+
+    @property
+    def account_roles(self) -> list[str]:
+        if not self.sts:
+            return []
+        roles = []
+        if self.sts.role_arn:
+            roles.append(self.sts.role_arn)
+        if self.sts.support_role_arn:
+            roles.append(self.sts.support_role_arn)
+        for instance_iam_role in (self.sts.instance_iam_roles or {}).values():
+            roles.append(instance_iam_role)
+        return roles
+
+    @property
+    def account_role_prefix(self) -> Optional[str]:
+        INSTALLER_ROLE_SUFFIX = "-Installer-Role"
+        for role in self.account_roles:
+            role_name = get_role_name_from_arn(role)
+            if role_name.endswith(INSTALLER_ROLE_SUFFIX):
+                return role_name[: -len(INSTALLER_ROLE_SUFFIX)]
```

Comment:
can directly use `removesuffix` like `return role_name.removesuffix(INSTALLER_ROLE_SUFFIX)`

Reply:
btw to express find first role with suffix or return None, can use `next`

```python
        return next(
            (
                role_name.removesuffix(INSTALLER_ROLE_SUFFIX)
                for role in self.account_roles
                if (role_name := get_role_name_from_arn(role))
                and role_name.endswith(INSTALLER_ROLE_SUFFIX)
            ),
            None,
        )
```

Reply:
i switched using removesuffix and also directly accesing the role_arn instead of searching for it first. that was some leftover